### ISSUE TITLE
Issues with empty datasets

### ISSUE THREAD

**Issue by gcphost** [deltaepsilon/firebase-paginator#1]
Hi

Thanks for the plugin.

I have an empty data set and this part of code causes a continuous loop due to the promise not resolving (I guess).

``` js
if (keys.length === pageSize + 1) {
```

`keys.length` is 0, no results, so it will never match pageSize 0 + 1

Changing it to

``` js
if (keys.length === pageSize + 1 || !keys.length) {
```

Seemed to have fixed it for me.

**Comment by gcphost** [comment]
Actually that still causes issues, needs bypass on 1 or less:

``` js
          if (keys.length <= 1) {
            isLastPage = true
          } else if (keys.length === pageSize + 1) {
```

**Comment by codemonkeynorth** [comment]
see also here @gcphost https://github.com/deltaepsilon/firebase-paginator/issues/4

it's also an issue if your data < pageSize

**Comment by deltaepsilon** [comment]
Fixed! 01be73d5c8794e93d05f2d6ea1f3cda59cc98778
